gravitee-apim-plugin/gravitee-apim-plugin-entrypoint/gravitee-apim-plugin-entrypoint-http-get/src/main/java/io/gravitee/plugin/entrypoint/http/get/HttpGetEntrypointConnector.java

```diff
@@ -0,0 +1,310 @@
+    static final String LIMIT_QUERY_PARAM = "limit";
+
+    static final Set<ConnectorMode> SUPPORTED_MODES = Set.of(ConnectorMode.SUBSCRIBE);
+
+    protected final HttpGetEntrypointConnectorConfiguration configuration;
+
+    @Override
+    public ListenerType supportedListenerType() {
+        return ListenerType.HTTP;
+    }
+
+    @Override
+    public Set<ConnectorMode> supportedModes() {
+        return SUPPORTED_MODES;
+    }
+
+    @Override
+    public int matchCriteriaCount() {
+        return 1;
+    }
+
+    @Override
+    public boolean matches(final ExecutionContext ctx) {
+        return (HttpMethod.GET == ctx.request().method());
+    }
+
+    @Override
+    public Completable handleRequest(final ExecutionContext ctx) {
+        return Completable.fromRunnable(
+            () -> {
+                ctx.putInternalAttribute(
+                    InternalContextAttributes.ATTR_INTERNAL_MESSAGES_LIMIT_DURATION_MS,
+                    configuration.getMessagesLimitDurationMs()
+                );
+
+                int messagesLimitCount = configuration.getMessagesLimitCount();
+                if (ctx.request().parameters().containsKey(LIMIT_QUERY_PARAM)) {
+                    String limit = ctx.request().parameters().getFirst(LIMIT_QUERY_PARAM);
+                    if (limit != null && !limit.isEmpty()) {
+                        messagesLimitCount = Math.min(messagesLimitCount, Integer.parseInt(limit));
+                    }
+                }
+                ctx.putInternalAttribute(InternalContextAttributes.ATTR_INTERNAL_MESSAGES_LIMIT_COUNT, messagesLimitCount);
+
+                if (ctx.request().parameters().containsKey(CURSOR_QUERY_PARAM)) {
+                    String cursor = ctx.request().parameters().getFirst(CURSOR_QUERY_PARAM);
+                    if (cursor != null && !cursor.isEmpty()) {
+                        ctx.putInternalAttribute(InternalContextAttributes.ATTR_INTERNAL_MESSAGES_RESUME_LASTID, cursor);
+                    }
+                }
+            }
+        );
+    }
+
+    @Override
+    public Completable handleResponse(final ExecutionContext ctx) {
+        String acceptHeader = ctx.request().headers().get(HttpHeaderNames.ACCEPT);
+        final String contentType = (acceptHeader == null || acceptHeader.isEmpty()) ? MediaType.TEXT_PLAIN : acceptHeader;
+        if (
+            contentType.equals(MediaType.APPLICATION_JSON) ||
+            contentType.equals(MediaType.APPLICATION_XML) ||
+            contentType.equals(MediaType.TEXT_PLAIN)
+        ) {
+            return Completable.fromRunnable(
+                () -> {
+                    ctx.response().headers().add(HttpHeaderNames.CONTENT_TYPE, contentType);
+                    ctx.response().chunks(messagesToBuffer(ctx, contentType));
+                }
+            );
+        } else {
+            return Completable.error(new IllegalArgumentException("Unsupported accept header: " + acceptHeader));
```

Comment:
I'm wondering if it would be more appropriate / logical to manage this error on the `handleRequest()` instead. WDYT ?